File: tcs_manager_frontend/src/service/UserService.js
import axios from "axios";

const USER_BASE_API_URL = "http://localhost:8080/api/v1/users/"

class UserService {
    getAllUsers() {
        return axios.get(USER_BASE_API_URL)
            .then(response => response.data)
    }

    getUserById(userId) {
        return axios.get(USER_BASE_API_URL + `${userId}`)
            .then(response => response.data)
    }

    updateUser(userId, user) {
        return axios.put(USER_BASE_API_URL + `${userId}`, user)
            .then(response => response.data)
    }

    partialUpdate(userId, user) {
        return axios.patch(USER_BASE_API_URL + `${userId}`, user)
            .then(response => response.data)
    }

    createUser(user) {
        return axios.post(USER_BASE_API_URL, user)
            .then(response => response.data)
    }

    deleteUser(userId) {
        return axios.delete(USER_BASE_API_URL + `${userId}`)
            .then(response => response.data)
    }
}

export default new UserService()

File: tcs_manager_system/src/main/java/com/tcsManager/tcsmanager/service/UserService.java
package com.tcsManager.tcsmanager.service;

import java.util.List;
import java.util.Map;

import com.tcsManager.tcsmanager.entity.User;
import com.tcsManager.tcsmanager.exception.UserNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * UserService
 */
public interface UserService {

    ResponseEntity<User> getUserById(long id) throws UserNotFoundException;

    ResponseEntity<User> loginUser(String userName, String userPassword) throws UserNotFoundException;

    ResponseEntity<User> createUser(User user);

    ResponseEntity<List<User>> getAllUsers();

    ResponseEntity<User> updateUser(long userId, User userUpdated) throws UserNotFoundException;

    ResponseEntity<User> partialUpdateUser(long userId, Map<Object, Object> fields) throws UserNotFoundException;

    ResponseEntity<?> deleteUser(long id) throws UserNotFoundException;

    // temporal, just for testing
    ResponseEntity<User> updatePassword(String userName, String userPassword) throws UserNotFoundException;
}
